### PR TITLE
Use socks5-proxy dialer when BOSH_ALL_PROXY has ssh+socks5:// schema

### PR DESCRIPTION
**Pull request by genevievelesperance**
Uses the socks5-proxy dialer when `BOSH_ALL_PROXY` has `ssh+socks5://` schema and the path to an ssh key for the jumpbox with `?private-key=`.

Reference Issue: https://github.com/cloudfoundry/bosh-cli/issues/324

There was an error running `dep ensure` to add the socks5-proxy package. @zachgersh (the dep wizard) determined that the path to `github.com/pivotal-golang/clock` needed to be changed to `code.cloudfoundry.org/clock` in a test.

This was the error:
```
~/go/src/github.com/cloudfoundry/bosh-utils |master ↑1 ✓|
gl $ dep ensure
ensure Solve(): No versions of github.com/pivotal-golang/clock met constraints:
	master: Could not introduce github.com/pivotal-golang/clock@master, as its subpackage github.com/pivotal-golang/clock/fakeclock does not contain usable Go code (*pkgtree.NonCanonicalImportRoot).. (Package is required by (root).)
	master: Could not introduce github.com/pivotal-golang/clock@master, as its subpackage github.com/pivotal-golang/clock/fakeclock does not contain usable Go code (*pkgtree.NonCanonicalImportRoot).. (Package is required by (root).)
direnv: error .envrc is blocked. Run `direnv allow` to approve its content.
```

### CODE REVIEW

File: httpclient/socksify.go [hunk 1]
```diff
@@ -18,6 +22,28 @@ func SOCKS5DialFuncFromEnvironment(origDialer DialFunc) DialFunc {
 		return origDialer
 	}
 
+	if strings.HasPrefix(allProxy, "ssh+") && strings.Contains(allProxy, "?private-key=") {
+		allProxy = strings.TrimPrefix(allProxy, "ssh+socks5://")
```
Comment (cppforlife) [inline httpclient/socksify.go]: can we use URL parse to grab all these pieces?

Comment (genevievelesperance) [reply httpclient/socksify.go]: Yup.

**Comment by zachgersh** [comment]
@genevievelesperance - changed the base of this PR to develop 👍

**Comment by genevievelesperance** [comment]
@zachgersh Added

**Pull request merged**